### PR TITLE
Add IpPermissionsFull Arg to Multiple EC2 Commands

### PR DESCRIPTION
**Pull request by johnnywilkes**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/EXPANDR-1852

## Description
We have a use case that requires an XSOAR automation that will do the following when an over-permissive AWS Security Group(SG) is found:
1. Determine what security group(s) are over-permissive and on what port
2. Copy the SG and remove the over-permissive part(s)
3. Apply the copied/modified SG back on the correct EC2 instance network adapter

This is challenging with the current EC2 integration before there are a lot of inputs for commands like "aws-ec2-authorize-security-group-ingress-rule."  It would be much easier to copy the rule as a whole using the "IpPermissions" array outputted from the "aws-ec2-describe-security-groups" command.  Effectively this is my addition in this PR: adding the *IpPermissionsFull* argument to the following commands so that we can more easily remove/copy security group rules: *aws-ec2-authorize-security-group-ingress-rule*, *aws-ec2-authorize-security-group-egress-rule*, *aws-ec2-revoke-security-group-ingress-rule*, and *aws-ec2-revoke-security-group-egress-rule*

## Screenshots
N/A

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [X] Documentation

### CODE REVIEW

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 1]
```diff
@@ -1609,12 +1619,15 @@ def revoke_security_group_egress_command(args, aws_client):
     kwargs = {
         'GroupId': args.get('groupId')
     }
+    if args.get('IpPermissionsFull', None):
+        IpPermissions = json.loads(args.get('IpPermissionsFull'))
+        kwargs['IpPermissions'] = IpPermissions
```
Comment (dansterenson) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1624]: You can assign the value to the variable inside the if statement. Also when using get() function, if the key does not exist it returns None by default.
Please do it everywhere.
```suggestion
    if IpPermissionsFull := args.get('IpPermissionsFull'):
        IpPermissions = json.loads(IpPermissionsFull)
        kwargs['IpPermissions'] = IpPermissions
```

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 2]
```diff
@@ -1609,12 +1619,15 @@ def revoke_security_group_egress_command(args, aws_client):
...
+        IpPermissions_dict = create_ip_permissions_dict(args)
+        UserIdGroupPairs_dict = create_user_id_group_pairs_dict(args)
 
-    IpPermissions_dict = create_ip_permissions_dict(args)
-    UserIdGroupPairs_dict = create_user_id_group_pairs_dict(args)
-
-    IpPermissions_dict['UserIdGroupPairs'] = [UserIdGroupPairs_dict]
-    kwargs['IpPermissions'] = [IpPermissions_dict]
+        IpPermissions_dict['UserIdGroupPairs'] = [UserIdGroupPairs_dict]
+        kwargs['IpPermissions'] = [IpPermissions_dict]
```
Comment (dansterenson) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1630]: Why here IpPermissions_dict is being appended to a list and in line 1624 not?

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 3]
```diff
@@ -1587,8 +1594,11 @@ def revoke_security_group_ingress_command(args, aws_client):
         role_session_duration=args.get('roleSessionDuration'),
     )
     kwargs = {'GroupId': args.get('groupId')}
-
-    kwargs.update(create_policy_kwargs_dict(args))
+    if args.get('IpPermissionsFull', None):
+        IpPermissions = json.loads(args.get('IpPermissionsFull'))
+        kwargs['IpPermissions'] = IpPermissions
```
Comment (dansterenson) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1599]:
```suggestion
    if IpPermissionsFull := args.get('IpPermissionsFull', None):
        IpPermissions = json.loads(IpPermissionsFull)
        kwargs['IpPermissions'] = IpPermissions
```

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 4]
```diff
@@ -1503,14 +1507,17 @@ def authorize_security_group_egress_command(args, aws_client):
...
-    IpPermissions = []
-    UserIdGroupPairs = []
-    IpPermissions_dict = create_ip_permissions_dict(args)
-    UserIdGroupPairs_dict = create_user_id_group_pairs_dict(args)
-
-    UserIdGroupPairs.append(UserIdGroupPairs_dict)
-    IpPermissions_dict.update({'UserIdGroupPairs': UserIdGroupPairs})  # type: ignore
-    IpPermissions.append(IpPermissions_dict)
+    if args.get('IpPermissionsFull', None):
+        IpPermissions = json.loads(args.get('IpPermissionsFull'))
```
Comment (dansterenson) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1511]:
```suggestion
    if IpPermissionsFull := args.get('IpPermissionsFull', None):
        IpPermissions = json.loads(IpPermissionsFull)
```

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py [hunk 5]
```diff
@@ -1476,17 +1477,20 @@ def authorize_security_group_ingress_command(args, aws_client):
         role_session_duration=args.get('roleSessionDuration'),
     )
     kwargs = {'GroupId': args.get('groupId')}
-    IpPermissions = []
-    UserIdGroupPairs = []
-    IpPermissions_dict = create_ip_permissions_dict(args)
-    UserIdGroupPairs_dict = create_user_id_group_pairs_dict(args)
+    if args.get('IpPermissionsFull', None):
+        IpPermissions = json.loads(args.get('IpPermissionsFull'))
```
Comment (dansterenson) [inline Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1481]:
```suggestion
    if IpPermissionsFull := args.get('IpPermissionsFull', None):
        IpPermissions = json.loads(IpPermissionsFull)
```

**Review by dansterenson — CHANGES_REQUESTED** [review]
Thanks for your contribution, Please see my notes.

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py (hunk 2 above)
Comment (johnnywilkes) [reply Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1630]: The assumption is that you would be passing it a list object already.  Example:
```
all_traffic_rule = """[{"IpProtocol": "-1", "IpRanges": [{"CidrIp": "0.0.0.0/0"}], "Ipv6Ranges": [], "PrefixListIds": [], "UserIdGroupPairs": []}]"""
return_results(demisto.executeCommand("aws-ec2-revoke-security-group-ingress-rule", {"groupId":'sg-0aa2f27bb524c2e01', "IpPermissionsFull":all_traffic_rule}))
```

**Comment by johnnywilkes** [comment]
@dansterenson, i made requested changes and answered your question.  Please let me know the next steps when possible.

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py (hunk 2 above)
Comment (dansterenson) [reply Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1630]: So maybe it should be documented in the argument description and README.

**Comment by dansterenson** [comment]
@johnnywilkes Answered your question above. In addition, please add a uni test to one of the commands which is using this additional argument you added. It is an XSOAR supported pack and it is important that future developers will know how this command is being used and what is the use case of this new argument. In this case, uni tests are great tools for that. In addition please add a short video demo to show the new use cases.

File: Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py (hunk 2 above)
Comment (johnnywilkes) [reply Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py:1630]: made changes in [add ippermissionsfull example](https://github.com/demisto/content/pull/25209/commits/df1a589027cd09dcda6a145316415c5e645fd46c)

**Comment by johnnywilkes** [comment]
@dansterenson , I added a unit test for `authorize_security_group_ingress`.  Will work on a quick demo video.

**Review by dansterenson — APPROVED** [review]
Great Job

**Pull request merged**